File: validpalindrome.java
public class Solution {
    static HashSet<Character> hs ;
    
    static {
        hs = new HashSet<Character>();
        for(char a = 'a' ; a <= 'z'; a++){
            hs.add(a);
        }
        for(char a = '0' ; a <= '9'; a++){
            hs.add(a);
        }
    }


    public boolean isPalindrome(String s) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(s.length()<=1){
            return true;
        }
        s = s.toLowerCase();
        int left = 0;
        int right = s.length()-1;
        while(left<=right){
            if(!hs.contains(s.charAt(left))){
                left ++;
                continue;
            }
            if(!hs.contains(s.charAt(right))){
                right--;
                continue;
            }
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            else{
                left++;
                right--;
            }
        }
        return true;
    }
    
    
}

File: cpp/ValidPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string sNew = "";
        for(int i=0;i<s.length();i++) {
            if(s[i]>='A'&&s[i]<='Z') {
                sNew.append(1, char(s[i]-'A'+'a'));
            } else if(s[i]>='a'&&s[i]<='z' || s[i]>='0'&&s[i]<='9') {
                sNew.append(1, s[i]);
            } 
        }
        int i=0;
        int j=sNew.length()-1;
        for(;i<j;i++,j--) {
            if(sNew[i]!=sNew[j]) {
                return false;
            }
        }
        return true;
    }
};
